tempodb/encoding/vparquet3/block_findtracebyid.go

```diff
@@ -251,6 +251,13 @@ func findTraceByID(ctx context.Context, traceID common.ID, meta *backend.BlockMe
 		return nil, errors.Wrap(err, "error reading row from backend")
 	}
 
+	if maxTraceSizeBytes > 0 {
+		estimatedSize := estimateMarshalledSizeFromTrace(tr)
+		if estimatedSize > maxTraceSizeBytes {
+			return nil, errors.Errorf("trace exceeds max size in the block. size: %d, max: %d", estimatedSize, maxTraceSizeBytes)
+		}
+	}
+
```

Comment:
Size check in `backendBlock` is missing in `vParquet` and `vParquet2`

Reply:
i'm ok with this. we are purposefully not backporting all changes to all previous parquet versions for sanity.